cfpb/consumerfinance.gov
# Add optional text introduction to tables

Add an optional text introduction to tables. See github.local/CFGOV/platform/issues/4547 for more details.

---




## Additions

- Text introduction block

## How to test this PR

1. Add a table to a page in Wagtail and confirm there's an optional plain text "text introduction" field in the table block as seen in the screenshot below
2. Add a text introduction to the table and confirm it renders like the screenshot below (above the table with a 30 px margin between text and table)
3. Confirm that tables without text introductions are unchanged; some good examples are `/data-research/research-reports/banks-overdraft-nsf-fee-revenue-declines-significantly-compared-to-pre-pandemic-levels/` and `/enforcement/payments-harmed-consumers/civil-penalty-fund/allocation-schedule/`

## Screenshots

Wagtail | Rendered
---- | ----
![table-block](https://github.com/cfpb/consumerfinance.gov/assets/1862695/eeb3da18-7153-431a-aa2c-d1145e7bb9dd) | ![table-intro](https://github.com/cfpb/consumerfinance.gov/assets/1862695/962a71dc-cb69-43a9-a7cd-7b8d956258c9) 

## Notes and todos

- Currently, if you add a table as the very first block in the content area of a page, there will be a 120px margin between the page's heading (e.g. text introduction) and the table. I chatted with @anselmbradford about what's going on there. Since it's unrelated to these changes (and a little complicated), I'm going to address that in a separate PR to keep this one simple.

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
